src/resolve/resolved-dns-rr.c

```diff
@@ -182,6 +182,23 @@ bool dns_resource_key_is_dnssd_ptr(const DnsResourceKey *key) {
                 dns_name_endswith(dns_resource_key_name(key), "_udp.local");
 }
 
+bool dns_resource_key_is_dnssd_two_label_ptr(const DnsResourceKey *key) {
+        assert(key);
+
+        /* Check if this is a PTR resource key used in Service Instance
+         * Enumeration as described in RFC6763 § 4.1, excluding selective
+         * service names described in RFC6763 § 7.1. */
+
+        if (key->type != DNS_TYPE_PTR)
+                return false;
+
+        const char *name = dns_resource_key_name(key);
+        if (dns_name_parent(&name) <= 0)
+                return false;
+
+        return dns_name_equal(name, "_tcp.local") || dns_name_equal(name, "_udp.local");
```

Comment:
Are you sure it can fail here though? Looked like the only failure is EINVAL, which I don't think is possible here.

Reply:
yeah, in its current implementation EINVAL is the only error we can see here, but we have this rule that if errors are ignored (which often is ok), then this should really be made clear via a `> 0` or `!= 0` check. i.e. relying on C's downgrade to bool logic should not be used, because this creates the impression this was actually a boolean, but it isn't.

Reply:
Ok, that makes sense. Thanks for the fix.